MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -246,62 +297,251 @@ int main(int argc, char * argv[])
-                std::wcout << "Please confirm the given package path is an .appx, .appxbundle, .msix, or .msixbundle file" << std::endl;
+                std::wcout << "Image successfully mounted!" << std::endl;
+                std::wcout << "To examine contents in File Explorer, press Win + R and enter the following: " << std::endl;
+                std::wcout << "\\\\?\\Volume{" << volumeIdString << "}" << std::endl;
+                std::wcout << std::endl;
+                std::wcout << "To unmount, run the following command: " << std::endl;
+                std::wcout << "msixmgr.exe -unmountimage -volumeid " << volumeIdString << " -filetype CIM" << std::endl;
                 std::wcout << std::endl;
-                return E_INVALIDARG;
             }
-            if (FAILED(hr))
+            else
             {
                 std::wcout << std::endl;
-                std::wcout << L"Failed with HRESULT 0x" << std::hex << hr << L" when trying to unpack " << packageFilePath << std::endl;
-                if (hr == static_cast<HRESULT>(MSIX::Error::CertNotTrusted))
+                std::wcout << "Please specify one of the following supported file types for the -MountImage command: {CIM}" << std::endl;
+                std::wcout << std::endl;
+                return ERROR_NOT_SUPPORTED;
+            }
+            return S_OK;
+        }
+        case OperationType::UnmountImage:
+        {
+            if (cli.GetFileType() == WVDFileType::CIM)
+            {
+                if (cli.GetVolumeId().empty())
                 {
-                    std::wcout << L"Please confirm that the certificate has been installed for this package" << std::endl;
+                    std::wcout << std::endl;
+                    std::wcout << "Please provide the id of the volume you would like to unmount using the -volumeId option" << std::endl;
+                    std::wcout << std::endl;
+                    return E_INVALIDARG;
                 }
-                else if (hr == static_cast<HRESULT>(MSIX::Error::FileWrite))
+
+                std::wstring volumeIdString = cli.GetVolumeId();
+                GUID volumeIdFromString;
+                if (UuidFromStringW((RPC_WSTR)(cli.GetVolumeId().c_str()), &volumeIdFromString) != RPC_S_OK)
                 {
-                    std::wcout << L"The tool encountered a file write error. If you are unpacking to a VHD, please try again with a larger VHD, as file write errors may be caused by insufficient disk space." << std::endl;
+                    std::wcout << std::endl;
+                    std::wcout << "Failed to convert specified volume id {" << volumeIdString << "}  to GUID" << std::endl;
+                    std::wcout << std::endl;
+                    return E_UNEXPECTED;
                 }
-                std::wcout << std::endl;
+
+                RETURN_IF_FAILED(MsixCoreLib::UnmountCIM(volumeIdFromString));

```

Comment:
>UnmountCIM [](start = 46, length = 10)

if this fails, nothing to tell user?

Reply:
doesn't say anything if it succeeded either.

---
In reply to: [473455726](https://github.com/microsoft/msix-packaging/pull/377#discussion_r473455726) [](ancestors = 473455726)